internal/discovery/collector/publishing_test.go

```diff
@@ -85,27 +88,79 @@ func (suite *PublishingTestSuite) TestCollectorClientPublishingSubscriptionDisco
 func (suite *PublishingTestSuite) TestCollectorClientPublishingSAPSystemDatabaseDiscovery() {
 	discoveredSAPSystem := mocks.NewDiscoveredSAPSystemDatabaseMock()
 
-	suite.runDiscoveryScenario(discoveryType, discoveredSAPSystem, func(requestBodyAgainstCollector string) {
+	suite.runDiscoveryScenario(sapSystemdiscovery, discoveredSAPSystem, func(requestBodyAgainstCollector string) {
 		suite.assertJSONMatchesJSONFileContent(helpers.GetFixturePath("discovery/sap_system/expected_published_sap_system_discovery_database.json"), requestBodyAgainstCollector)
 	})
 }
 
 func (suite *PublishingTestSuite) TestCollectorClientPublishingSAPSystemApplicationDiscovery() {
 	discoveredSAPSystem := mocks.NewDiscoveredSAPSystemApplicationMock()
 
-	suite.runDiscoveryScenario(discoveryType, discoveredSAPSystem, func(requestBodyAgainstCollector string) {
+	suite.runDiscoveryScenario(sapSystemdiscovery, discoveredSAPSystem, func(requestBodyAgainstCollector string) {
 		suite.assertJSONMatchesJSONFileContent(helpers.GetFixturePath("discovery/sap_system/expected_published_sap_system_discovery_application.json"), requestBodyAgainstCollector)
 	})
 }
 
 func (suite *PublishingTestSuite) TestCollectorClientPublishingSAPSystemDiagnosticsDiscovery() {
 	discoveredSAPSystem := mocks.NewDiscoveredSAPSystemDiagnosticsMock()
 
-	suite.runDiscoveryScenario(discoveryType, discoveredSAPSystem, func(requestBodyAgainstCollector string) {
+	suite.runDiscoveryScenario(sapSystemdiscovery, discoveredSAPSystem, func(requestBodyAgainstCollector string) {
 		suite.assertJSONMatchesJSONFileContent(helpers.GetFixturePath("discovery/sap_system/expected_published_sap_system_discovery_diagnostics.json"), requestBodyAgainstCollector)
 	})
 }
 
+func (suite *PublishingTestSuite) TestCollectorClientPublishingAWSCloudDiscoveryWithMetadata() {
+	discoveredCloudInstance := cloud.Instance{
+		Provider: cloud.AWS,
+		Metadata: &cloud.AWSMetadata{
+			AmiID:              "ami-123456",
+			BlockDeviceMapping: map[string]string{"device": "volume"},
+			InstanceID:         "i-123456",
+			InstanceType:       "t2.micro",
+			Placement: cloud.Placement{
+				AvailabilityZone: "us-west-2a",
+				Region:           "us-west-2",
+			},
+		},
+	}
+
+	suite.runDiscoveryScenario(cloudDiscovery, discoveredCloudInstance, func(requestBodyAgainstCollector string) {
+		suite.assertJSONMatchesJSONFileContent(helpers.GetFixturePath("discovery/aws/collector_payloads/expected_published_cloud_discovery_with_metadata.json"), requestBodyAgainstCollector)
+	})
+}
+
+func (suite *PublishingTestSuite) TestCollectorClientPublishingAWSCloudDiscoveryWithoutMetadata() {
```

Comment:
nitpick: maybe you could include this test with the final one. At the end, they look the same, don't they?

Reply:
Alright a190a0ac60953af98fa5cd562f9be68202ddb31f